In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Dataset Representation
### CSMODELT3AY2021In this section of the notebook, you must fulfill the following:
- State a brief description of the dataset.
- Provide  a  description  of  the  collection  process  executed  to  build  the  dataset. Discuss  the  implications  of  the  data  collection  method  on  the  generated conclusions and insights. Note that you may need to look at the relevant sources related to the dataset to be able to provide the necessary information for this part of the project.
- Describe the structure of the dataset file. In the dataset file, what does each row and  column  represent?  How many observations  are  there  in  the  dataset? How many variables are there in the dataset? If thedataset is composed of different files  that  you  will  combine  in  the  succeeding  steps,  you  need  to  describe  the structure and the contents of each file.
- Discuss the variables in each dataset file. What does each variable represent? In this section, all variables, even those which are not used for the study, should be described to the reader. The purpose of each variable in the dataset should be clear to the reader of the notebook without having to go through an external link.

### Brief Description of the Dataset
The dataset contains 30 years (1990-2020) worth of data collected by the World Food Programme (WFP) from 99 different countries which contains the Global Food Prices. The data includes information on country, city, market, price of good in local currency, quantity of good, and month and year recorded.

### Collection Process
** insert here **

Source: https://www.kaggle.com/salehahmedrony/global-food-prices

In [2]:
df = pd.read_csv('wfp_food_prices_database.csv')

c:\users\chris\appdata\local\programs\python\python39\lib\site-packages\IPython\core\interactiveshell.py:3165: DtypeWarning: Columns (3) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [3]:
df.head()

adm0_id    adm0_name  adm1_id   adm1_name  mkt_id  mkt_name  cm_id  \
0      1.0  Afghanistan      272  Badakhshan     266  Fayzabad     55   
1      1.0  Afghanistan      272  Badakhshan     266  Fayzabad     55   
2      1.0  Afghanistan      272  Badakhshan     266  Fayzabad     55   
3      1.0  Afghanistan      272  Badakhshan     266  Fayzabad     55   
4      1.0  Afghanistan      272  Badakhshan     266  Fayzabad     55   

          cm_name  cur_id cur_name  pt_id pt_name  um_id um_name  mp_month  \
0  Bread - Retail     0.0      AFN     15  Retail      5      KG         1   
1  Bread - Retail     0.0      AFN     15  Retail      5      KG         2   
2  Bread - Retail     0.0      AFN     15  Retail      5      KG         3   
3  Bread - Retail     0.0      AFN     15  Retail      5      KG         4   
4  Bread - Retail     0.0      AFN     15  Retail      5      KG         5   

   mp_year  mp_price  mp_commoditysource  
0     2014      50.0                 NaN  
1     2014      50.0                 NaN  
2     2014      50.0                 NaN  
3     2014      50.0                 NaN  
4     2014      50.0                 NaN

### Structure of the Dataset
** insert here **

In [4]:
df.columns

Index(['adm0_id', 'adm0_name', 'adm1_id', 'adm1_name', 'mkt_id', 'mkt_name',
       'cm_id', 'cm_name', 'cur_id', 'cur_name', 'pt_id', 'pt_name', 'um_id',
       'um_name', 'mp_month', 'mp_year', 'mp_price', 'mp_commoditysource'],
      dtype='object')

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560551 entries, 0 to 1560550
Data columns (total 18 columns):
 #   Column              Non-Null Count    Dtype  
---  ------              --------------    -----  
 0   adm0_id             1560551 non-null  float64
 1   adm0_name           1560551 non-null  object 
 2   adm1_id             1560551 non-null  int64  
 3   adm1_name           1089223 non-null  object 
 4   mkt_id              1560551 non-null  int64  
 5   mkt_name            1560551 non-null  object 
 6   cm_id               1560551 non-null  int64  
 7   cm_name             1560551 non-null  object 
 8   cur_id              1560551 non-null  float64
 9   cur_name            1560551 non-null  object 
 10  pt_id               1560551 non-null  int64  
 11  pt_name             1560551 non-null  object 
 12  um_id               1560551 non-null  int64  
 13  um_name             1560551 non-null  object 
 14  mp_month            1560551 non-null  int64  
 15  mp_year        

### Variables of the Dataset
There are 18 unique variables in the dataset with 1560551 entries.
The variables in the dataset are as follows:

| Column Name        | Full Name                          | Definition        |
| ------------------ | ---------                          | ----------------- |
| adm0_id            | country id                         | ** insert here ** |
| adm0_name          | country name                       | ** insert here ** |
| adm1_id            | locality id                        | ** insert here ** |
| adm1_name          | locality name                      | ** insert here ** |
| mkt_id             | market id                          | ** insert here ** |
| mkt_name           | market name                        | ** insert here ** |
| cm_id              | commodity purchase id              | ** insert here ** |
| cm_name            | commodity purchased                | ** insert here ** |
| cur_id             | currency  id                       | ** insert here ** |
| cur_name           | name of the currency               | ** insert here ** |
| pt_id              | market type id                     | ** insert here ** |
| pt_name            | market type                        | ** insert here ** |
| um_id              | measurement id                     | ** insert here ** |
| um_name            | unit of goods measurement          | ** insert here ** |
| mp_month           | month recorded                     | ** insert here ** |
| mp_year            | year recorded                      | ** insert here ** |
| mp_price           | price paid                         | ** insert here ** |
| mp_commoditysource | source supplying price information | ** insert here ** |

## Data Cleaning
For each used variable, check for the following and, if needed, perform data cleaning:
- There are multiple representations of the same categorical value
- The datatype of the variable is incorrect.
- Some values are set to default values of the variable.
- There are missing data.
- There are duplicate data.
- The formatting of the values is inconsistent.  

**Note**: No need to clean all variables. Clean only the variables utilized in the study.

**TODO**: 
- [ ] Remove the ID columns (since those are not needed)
- [ ] Check for null values

In [55]:
missingVars = df.columns[df.isnull().any()].tolist()

In [56]:
df.loc [df['High total cholesterol'].isnull(),'High total cholesterol'] == "None"
df.loc [df['Outdoor air pollution'].isnull(),'Outdoor air pollution'] == "None"

3751    False
Name: Outdoor air pollution, dtype: bool

In [57]:
df.loc [df['High total cholesterol'] == "None", 'High total cholesterol'].shape

(0,)

## Exploratory Data Analysis
In this section of the notebook, you must fulfill the following:
- Identify 3 interesting exploratory data analysis questions.Properly state the questions in the notebook.
- Answer the EDA questionsusing both:
    - Numerical Summaries – measures of central tendency, measures of dispersion, and correlation
    - Visualization – Appropriate  visualization should be used. Each visualization should be accompanied by a brief explanation.
- To emphasize, both numerical summary and visualization should be present to answer each question. The whole process should be supported withverbose textual descriptions of your proceduresand findings.

## Research Question
Come up with research questions to answer using the dataset.The research questions should arise from the exploratory data analysis.
- The first research question should be answerable by performing statistical inference on means.
- The second research question should be answerable by performing statistical inference on categorical data.
- The research questions should be within the scope of the dataset.
- For each research question, you must indicate its importance and significance to the community.

## Statistical Inference
In this section of the notebook, please take note of the following:
- If needed, you may perform preprocessing techniques to transform the data to the appropriate representation before performing statistical inference to answer the research question. This may include binning, log transformations, conversion to one-hot encoding, normalization, standardization, interpolation, truncation, and feature engineering. You may also need to check and prove if the data is from a normal distribution to perform some statistical inference techniques.
- The techniques used to answer the research questionare limited to statistical inference methods discussed in class. 
- The technique that you will apply should be appropriate to answer the research question.

## Insights and Conclusions
Clearly  state  your  insights  and  conclusions  from  the  data  to  answer  each  research question you have defined. Make sure that all conclusions are backed up with statistical evidence.